src/stan/callbacks/unique_stream_writer.hpp

```diff
@@ -66,17 +67,34 @@ class unique_stream_writer final : public writer {
    *
    * @param[in] state Values in a std::vector
    */
-  void operator()(const std::vector<double>& state) { write_vector(state); }
+  void operator()(const std::vector<double>& state) {
+    if (output_ == nullptr)
+      return;
+    write_vector(state);
+  }
 
-  /**
-   * Writes the comment_prefix to the stream followed by a newline.
-   */
-  void operator()() {
+  void operator()(const std::tuple<Eigen::VectorXd, Eigen::VectorXd>& state) {
     if (output_ == nullptr)
       return;
-    *output_ << comment_prefix_ << std::endl;
+    Eigen::IOFormat CommaInitFmt(Eigen::StreamPrecision, Eigen::DontAlignCols,
+                                 ", ", "", "", "\n", "", "");
+    *output_ << std::get<0>(state).transpose().eval();
+    *output_ << std::get<1>(state).transpose().eval();
+  }
+
+  void operator()(const Eigen::MatrixXd& states) {
```

Comment:
This overload needs a docstring (although this isn't part of `stan::io::callbacks::writer`, so maybe it isn't used?)

Is there a story behind the name `states`?

Reply:
gone - same as above - early pathfinder version, checked in by mistake.